manifests/plugin/container_gateway.pp

```diff
@@ -22,13 +36,36 @@
   Boolean $enabled = true,
   Foreman_proxy::ListenOn $listen_on = 'https',
   Stdlib::HTTPUrl $pulp_endpoint = "https://${facts['networking']['fqdn']}",
+  Enum['postgres', 'sqlite'] $database_backend = 'postgres',
   Stdlib::Absolutepath $sqlite_db_path = '/var/lib/foreman-proxy/smart_proxy_container_gateway.db',
   Optional[Integer] $sqlite_timeout = undef,
+  Boolean $manage_postgresql = true,
+  Optional[Stdlib::Host] $postgresql_host = undef,
+  Optional[Stdlib::Port] $postgresql_port = undef,
+  String $postgresql_database = 'container_gateway',
+  Optional[String[1]] $postgresql_user = undef,
+  Optional[String] $postgresql_password = undef
 ) {
   foreman_proxy::plugin::module { 'container_gateway':
     version   => $version,
     enabled   => $enabled,
     feature   => 'Container_Gateway',
     listen_on => $listen_on,
   }
+
+  if $manage_postgresql and $database_backend == 'postgres' {
+    include postgresql::server
+    $_postgresql_user = pick($postgresql_user, $foreman_proxy::user)
+    postgresql::server::db { $postgresql_database:
+      user     => $_postgresql_user,
+      password => postgresql::postgresql_password(
+        $_postgresql_user,
+        $postgresql_password.lest || {
+          extlib::cache_data('container_gateway_cache_data', 'db_password', extlib::random_password(32))
+        }
+      ),
```

Comment:
I don't think we even need to set a password at all when we use a unix socket. I'm fine merging this now as is, but probably worth investigating.